CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/gui/JMultiChoiceComboBoxMenuNGTest.java

```diff
@@ -0,0 +1,369 @@
+        System.out.println("removeSelectedItem");
+        String s1 = "Item 1";
+        String s2 = "Item 3";
+        String s3 = "Item 4";
+        String[] selectedItems = {s1, s2, s3};
+        JMultiChoiceComboBoxMenu instance = new JMultiChoiceComboBoxMenu("Text", items);
+
+        instance.addSelectedItems(s1, s2, s3);
+        Set itemsbeforeRemove = instance.getSelectedItems();
+        assertEquals(itemsbeforeRemove.size(), 3);
+        assertEqualsNoOrder(itemsbeforeRemove.toArray(), selectedItems);
+
+        instance.removeSelectedItem(s2);
+        Set resultSet = instance.getSelectedItems();
+        assertEquals(resultSet.size(), 2);
+        assertEqualsNoOrder(resultSet.toArray(), new String[]{s1, s3});
+    }
+
+    /**
+     * Test of removeSelectedItems method, of class JMultiChoiceComboBoxMenu.
+     */
+    @Test
+    public void testRemoveSelectedItems() {
+        System.out.println("removeSelectedItems");
+        String s1 = "Item 1";
+        String s2 = "Item 3";
+        String s3 = "Item 4";
+        String[] selectedItems = {s1, s2, s3};
+        JMultiChoiceComboBoxMenu instance = new JMultiChoiceComboBoxMenu("Text", items);
+
+        instance.addSelectedItems(s1, s2, s3);
+        Set itemsbeforeRemove = instance.getSelectedItems();
+        assertEquals(itemsbeforeRemove.size(), 3);
+        assertEqualsNoOrder(itemsbeforeRemove.toArray(), selectedItems);
+
+        instance.removeSelectedItems(s2, s3);
+        Set resultSet = instance.getSelectedItems();
+        assertEquals(resultSet.size(), 1);
+        assertEquals(resultSet.toArray(), new String[]{s1});
+    }
+
+    /**
+     * Test of clearSelection method, of class JMultiChoiceComboBoxMenu.
+     */
+    @Test
+    public void testClearSelection() {
+        System.out.println("clearSelection");
+        String s1 = "Item 1";
+        String s2 = "Item 3";
+        String s3 = "Item 4";
+        String[] selectedItems = {s1, s2, s3};
+        JMultiChoiceComboBoxMenu instance = new JMultiChoiceComboBoxMenu("Text", items);
+
+        instance.addSelectedItems(s1, s2, s3);
+        Set itemsbeforeRemove = instance.getSelectedItems();
+        assertEquals(itemsbeforeRemove.size(), 3);
+        assertEqualsNoOrder(itemsbeforeRemove.toArray(), selectedItems);
+
+        instance.clearSelection();
+        Set resultSet = instance.getSelectedItems();
+        assertEquals(resultSet.size(), 0);
+    }
+
+    /**
+     * Test of addSelectionListener method, of class JMultiChoiceComboBoxMenu.
+     */
+    @Test
+    public void testAddSelectionListener() {
+        System.out.println("addSelectionListener");
+        ListSelectionListener listSelectionListenerMock = mock(ListSelectionListener.class);
+        JMultiChoiceComboBoxMenu instance = mock(JMultiChoiceComboBoxMenu.class);
+        instance.addSelectionListener(listSelectionListenerMock);
+        verify(instance, times(1)).addSelectionListener(listSelectionListenerMock);
```

Comment:
can you retrieve the selection listener to check it at all?
It seems like the test really verifies that the above call was executed and would probably always pass

Reply:
Agree with @aldebaran30701. It seems in the current state this is simply testing the function executed rather than checking it executed as intended

Reply:
I guess Verify is still a valid way of testing a void method. All I can think of is changing the modifier of `Set<ListSelectionListener> listeners` to protected or adding a getter. Not sure it's worth the effort because this method just adds the listener in the collection.

Reply:
Usually if it's one line I wouldn't say to worry, but considering the test case is already set up, it wouldn't hurt to change it to protected given it's useful to test

Reply:
@Auriga2 verify can be used to check calls made within a function you're testing but shouldn't really be used for the function itself. Otherwise it doesn't actually matter what the function implementation is (e.g. a blank implementation would pass this test, even though that's definitely a bug). 

If you were to check this, I would add a getter rather than change the modifier in this instance. If you feel the method is too trivial to check though, it may be worth omitting this unit test all together. Worth noting though that you don't appear to use the function for any of the other unit tests so I think it may still be worth having the basic test in this instance

Reply:
Thanks, it was my misunderstanding. Fixed now.